smart-common/src/main/java/org/smartdata/model/ActionInfo.java

```diff
@@ -15,139 +15,101 @@
 
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder(setterPrefix = "set")
 public class ActionInfo {
   // Old file id
   public static final String OLD_FILE_ID = "-oid";
+
   private long actionId;
   private long cmdletId;
   private String actionName;
   private Map<String, String> args;
-  private String result;
-  private String log;
+  @lombok.Builder.Default
+  private String result = "";
+  @lombok.Builder.Default
+  private String log = "";
 
   // For action set flexibility
-  private String execHost;
   private boolean successful;
   private long createTime;
   private boolean finished;
   private long finishTime;
-
   private float progress;
-
-  public ActionInfo() {
-    this.result = "";
-    this.log = "";
-  }
+  private String execHost;
 
   public ActionInfo(long actionId, long cmdletId, String actionName,
-      Map<String, String> args, String result, String log,
-      boolean successful, long createTime, boolean finished,
-      long finishTime, float progress) {
-    this.actionId = actionId;
-    this.cmdletId = cmdletId;
-    this.actionName = actionName;
-    this.args = args;
-    this.result = result;
-    this.log = log;
-    this.execHost = "";
-    this.successful = successful;
-    this.createTime = createTime;
-    this.finished = finished;
-    this.finishTime = finishTime;
-    this.progress = progress;
-  }
-
-  public long getActionId() {
-    return actionId;
+                    Map<String, String> args, String result, String log,
+                    boolean successful, long createTime, boolean finished,
+                    long finishTime, float progress) {
+    this(actionId, cmdletId, actionName,
+        args, result, log, successful, createTime,
+        finished, finishTime, progress, "");
   }
 
-  public void setActionId(long actionId) {
-    this.actionId = actionId;
+  public String getActionText() {
+    return actionToString(actionName, args);
   }
 
-  public long getCmdletId() {
-    return cmdletId;
-  }
-
-  public void setCmdletId(long cmdletId) {
-    this.cmdletId = cmdletId;
-  }
-
-  public String getActionName() {
-    return actionName;
-  }
-
-  public void setActionName(String actionName) {
-    this.actionName = actionName;
-  }
-
-  public Map<String, String> getArgs() {
-    return args;
-  }
-
-  public void setArgs(Map<String, String> args) {
-    this.args = args;
-  }
-
-  public String getArgsJsonString() {
-    Gson gson = new Gson();
-    return gson.toJson(args);
-  }
-
-  public void setArgsFromJsonString(String jsonArgs) {
-    Gson gson = new Gson();
-    args = gson.fromJson(jsonArgs,
-        new TypeToken<Map<String, String>>() {
-        }.getType());
+  public ActionSource getSource() {
+    boolean hasRuleIdArg = Optional.ofNullable(args)
+        .filter(map -> map.containsKey(CmdletDescriptor.RULE_ID))
+        .isPresent();
+    return hasRuleIdArg ? ActionSource.RULE : ActionSource.USER;
```

Comment:
You can continue optional stream: `map(map -> ActionSource.RULE).orElse(ActionSource.USER)`